### ISSUE TITLE
Copy @JsonDeserialize from Type methods to generated Builder methods

### ISSUE THREAD

**Issue by mcliedtke** [immutables/immutables#546]
Hello,
I use Immutables in combination with Jackson quite a bit, and I recently started using the builder as the mechanism for deserialization of the immutable types (annotating types with `@JsonDeserialize(builder=ImmutableXxxx.Builder.class)`). I ran into an issue when methods use custom deserialization (i.e. the methods are also annotated with `@JsonDeserialize(...)`). It seems these custom deserialization annotations are not copied to the generated builder and errors occur during deserialization. 

Below is an example illustrating the issue I am having. I have successfully resolved the issue by copying the `@JsonDeserialize(converter=IntValueOverrideConverter.StringDeserializer.class)` to the generated builder. This is a simple example with a vanilla object, but I imagine the same solution will work with methods returning container types like `Optional<>`, `List<>` etc... 

In contrast, deserializing types using `@JsonDeserialize(as=ImmutableXxxx.class)` seems to work fine because Jackson seems to be able to associate the setter and getter methods of the JSON type back with the parent method that has the original `@JsonDeserialize(...)` annotation. 

For my usecase, it's desirable to use the builder method to avoid issues with duplicate deserialization (#461). (Which might be why the `@JsonDeserialize(as=ImmutableXxxx.class)` ends up working?). Does copying the `@JsonDeserialize(...)` to the builder methods seem like a feature that could be added?

BuilderDeserializationTest.java
```java
import org.immutables.value.Value;
import org.junit.Assert;
import org.junit.Test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.util.StdConverter;

public class BuilderDeserializationTest {
    @Value.Immutable
    public abstract static class IntValueClass {

        @JsonValue
        public abstract Integer getValue();
        
        @JsonCreator
        public static IntValueClass of(final Integer value) {
            return ImmutableIntValueClass.builder().value(value).build();
        }
    }
    
    @Value.Immutable
    @JsonDeserialize(as=ImmutableWorkingType.class)
    public static abstract class WorkingType {
        
        @JsonProperty("int_value_property")
        @JsonSerialize(converter=IntValueOverrideConverter.StringSerializer.class)
        @JsonDeserialize(converter=IntValueOverrideConverter.StringDeserializer.class)
        public abstract IntValueClass getIntValue();
    }
    
    @Value.Immutable
    @JsonDeserialize(builder=ImmutableBrokenType.Builder.class)
    public static abstract class BrokenType {
        
        @JsonProperty("int_value_property")
        @JsonSerialize(converter=IntValueOverrideConverter.StringSerializer.class)
        @JsonDeserialize(converter=IntValueOverrideConverter.StringDeserializer.class)
        public abstract IntValueClass getIntValue();
    }
    
    public static class IntValueOverrideConverter {
        public static class StringSerializer extends StdConverter<IntValueClass, String> {
            @Override
            public String convert(final IntValueClass value) {
                return value.getValue().toString();
            }
        }
        
        public static class StringDeserializer extends StdConverter<String, IntValueClass> {
            @Override
            public IntValueClass convert(final String value) {
                return IntValueClass.of(Integer.parseInt(value));
            }
        }
    }
    
    private static final ObjectMapper MAPPER = new ObjectMapper();
    private static final IntValueClass INT_VALUE = IntValueClass.of(1);
    
    @Test
    public void workingTest() throws Exception {
        final WorkingType someType = ImmutableWorkingType.builder()
                .intValue(INT_VALUE)
                .build();

        final String serialized = MAPPER.writeValueAsString(someType);
        Assert.assertEquals("{\"int_value_property\":\"1\"}", serialized);
        
        //Succeeds deserializing, uses IntValueOverrideConverter.StringDeserializer
        final WorkingType someType2 = MAPPER.readValue(serialized, WorkingType.class);
        //Succeeds deserializing, uses IntValueOverrideConverter.StringDeserializer

        Assert.assertEquals(someType, someType2);
    }
    
    @Test
    public void brokenTest() throws Exception {
        final BrokenType someType = ImmutableBrokenType.builder()
                .intValue(INT_VALUE)
                .build();
        
        final String serialized = MAPPER.writeValueAsString(someType);
        Assert.assertEquals("{\"int_value_property\":\"1\"}", serialized);
        
        //Throws Exception on deserialization, cannot construct IntValueClass from string value
        final BrokenType someType2 = MAPPER.readValue(serialized, BrokenType.class);
        //Throws Exception on deserialization, cannot construct IntValueClass from string value
        
        Assert.assertEquals(someType, someType2); //unreached
    }
}
```

pom.xml
```xml
<project xmlns="http://maven.apache.org/POM/4.0.0" 
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">

  <modelVersion>4.0.0</modelVersion>

  <groupId>com.test.example</groupId>
  <artifactId>quickstart</artifactId>
  <version>0.0.1-SNAPSHOT</version>
  <packaging>jar</packaging>

  <name>quickstart</name>
  <url>http://maven.apache.org</url>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>

  <dependencies>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>4.12</version>
    </dependency>
    <dependency>
    	<groupId>org.immutables</groupId>
    	<artifactId>value</artifactId>
    	<version>2.4.3</version>
    	<scope>provided</scope>
    </dependency>
    <dependency>
    	<groupId>com.fasterxml.jackson.core</groupId>
    	<artifactId>jackson-databind</artifactId>
    	<version>2.8.6</version>
    </dependency>
  </dependencies>
</project>
```

**Comment by elucash** [comment]
Thank you for reporting this! Fixed in master and will be released in 2.4.4. (I've included the test provided to the codebase if you don't mind)

**Comment by mcliedtke** [comment]
Awesome, thank you for the quick fix! You're welcome to use the test.

**Comment by nkvoll** [comment]
Ran into this issue on `2.4.3` today, and can confirm it's working on `2.4.4-SNAPSHOT`. What a lucky timing for me!

Is there an ETA for a non-snapshot release for `2.4.4`?

**Comment by elucash** [comment]
2-3 days: couple of small issues to fix & then release

**Comment by mimi89999** [comment]
@edreyer So when can we expect `2.4.4` to be released as stable?

**Comment by elucash** [comment]
my appologies for the long wait, released in 2.4.4 thank you!
